typescript/cli/src/send/transfer.ts

```diff
@@ -75,31 +74,37 @@ export async function sendTestTransfer({
     });
   }
 
-  if (tokenType === TokenType.collateral) {
-    const collateralRouter = HypERC20Collateral__factory.connect(
-      routerAddress,
-      multiProvider.getProvider(origin),
-    );
-    const tokenAddress = await collateralRouter.wrappedToken();
+  // TODO: move to SDK token router app
+  // deduce TokenType
+  // 1. decimals() call implies synthetic
+  // 2. wrappedToken() call implies collateral
+  // 3. if neither, it's native
+  let tokenAddress: Address | undefined;
+  const provider = multiProvider.getProvider(origin);
+  try {
```

Comment:
nit: not a big fan of the nested try catch logic here but I don't know of any other alternatives

Reply:
me neither tbh